Rendering dummy data from database

dcheng1290 opened the pull request:
Please review my react components for my front end. 

My addCommentForm is a POST method that has not been implemented successfully yet. It is a component that provide me the comment form but it cannot insert data yet, so the functionality can be ignored. 

I added additional comments into my seed file so that each recipe will have 4 comments by default as of now. The CommentList.jsx is able to show all 4 comments with the appropriate fields. The App.jsx is able to show the non functional post form box and also the comments from database.

Thanks!

Commit:
React Components Rendering Dummy Data

Commit:
Rendering data from database

dcheng1290 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [PR David's initial front end components](https://trello.com/c/gz27GlJW/40-pr-davids-initial-front-end-components)

careyl96 commented on server/index.js:
```diff
@@ -14,8 +14,8 @@ app.post('/recipe', (req, res) => {
 
-app.get('/recipe', (req, res) => {
-  return Comments.getRecipe()
+app.get('/recipe/:id', (req, res) => {
```
the route should be /recipe/:id/comments; there are many different modules for any given recipe and the route for the API call should reflect the information you're intending to get.

careyl96 commented on database/comments.js:
```diff
@@ -16,15 +16,15 @@ const commentSection = new mongoose.Schema({
-// getRecipe function
-const getRecipe = (recipeName) => {
+
+const getComment = (recipeID) => {
```
Consider renaming getComment to getCommentData or getComments (plural) since you are, if I'm not mistaken, grabbing a -collection- of data containing more information than just the comments themselves

dcheng1290 replied:
Updated to getComments for all files

careyl96 commented on database/seed.js:
```diff
@@ -19,6 +19,24 @@ const sampleData = () => {
           fullName: faker.name.findName(),
           created_At: faker.date.past(),
           image: faker.image.avatar(),
```
rename image to avatar/userimage/profileimage; image is too general

dcheng1290 replied:
Updated image to userImage for better name

careyl96 commented on client/src/App.jsx:
```diff
@@ -2,45 +2,50 @@ import React from 'react';
 import AddCommentForm from './AddCommentForm.jsx';
 import CommentList from './CommentList.jsx';
 import axios from 'axios';
```
add dependencies above components

dcheng1290 replied:
Moved axios above components

careyl96 commented on client/src/App.jsx:
```diff
@@ -2,45 +2,50 @@ import React from 'react';
-    
+  getComment(id) {
+    axios.get(`/recipe/${id}`)
+    .then(comment => this.setState({comments: comment.data[0].comments}))
```
I'm looking at the format of the data you receive when making an API call and it is very confusing; it looks like a get request to the API gives you an array consisting of a single object which consists an array of comments. The comment key is unnecessary and creates confusion. Instead, your data should simply be an array of objects since your route (which you should change to recipe/:id/comments) would already make it clear to anyone reading your code that you are going to be receiving a collection of comments.

careyl96 commented on server/index.js:
```diff
@@ -14,8 +14,8 @@ app.post('/recipe', (req, res) => {
-app.get('/recipe', (req, res) => {
-  return Comments.getRecipe()
+app.get('/recipe/:id', (req, res) => {
+  return Comments.getComment(req.params.id)
```
we aren't returning anything here; getComment is an asynchronous function and the information we need to send back to the client is information you receive after the promise resolves.

dcheng1290 replied:
Removed return as there is no need for it

careyl96 commented on client/src/CommentListItem.jsx:
```diff
@@ -1,12 +1,13 @@
 import React from 'react';
 
 const CommentListItem = (props) => (
```
this component can just be called Comment

dcheng1290 replied:
Updated CommentListItem to just Comment

careyl96 commented on database/comments.js:
```diff
@@ -16,15 +16,15 @@ const commentSection = new mongoose.Schema({
 
```
I think this is more of an index.js file than a comments.js file. The primary usage of this file seems to be initializing the database schema and models, not very related to comments

dcheng1290 replied:
I have moved over most of comment.js's code into index.js which is the primary file for initializing database and left the helper function in comment.js

careyl96 commented on client/src/App.jsx:
```diff
@@ -2,45 +2,50 @@ import React from 'react';
+        <section className="section">
+          <div className="tips">
+            <h3>Tips from Head Chefs</h3> 
+            <div><b>{this.state.comments.length} Comments</b></div>
```
remove \<b> tags; don't style within html

dcheng1290 replied:
Removed <b> tags

careyl96 commented on client/src/App.jsx:
```diff
@@ -2,45 +2,50 @@ import React from 'react';
 
-
+// add comment function is still in progress *stretch goal
   addComment(comment) {
```
rename to postComment

dcheng1290 replied:
Renamed to postComment

careyl96 reviewed (changes requested):
fix some syntax errors, fix API routes, modify schema, rename some variables, functions, and components. Also, break up pull request into smaller parts next time!

Commit:
Renamed CommentListItem component to Comment

Commit:
Updated variable name getComment to getComments, image field to userImage

Commit:
Updated index.js to be more of a db file, separated out helper functions

Commit:
Updated method name for addComment

Commit:
Updated App.jsx

dcheng1290 committed:
Merge branch 'master' into ReactComponents

dcheng1290 commented:
Thanks for the constructive feedbacks Carey! I have updated a lot of the naming issues, changed the database's index.js into more like a index.js while leaving helper function in comments.js. I also provided additional comments onto your reviews.
